src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/CompositeMLDsa/CompositeMLDsaFactoryTests.cs

```diff
@@ -158,69 +158,44 @@ static byte[] CreateKeyWithVersion(int? version)
 
                 return ComposeKeys(
                     MLDsaTestsData.IetfMLDsa65.PrivateSeed,
-                    WriteECPrivateKey(version, ecdsaKey.D, ecdsaKey.Curve.Oid.Value, ecdsaKey.Q));
+                    WriteECPrivateKey(version, ecdsaKey.D, oid: null, point: null));
             }
         }
 
         [Fact]
         public static void ImportBadPrivateKey_ECDsa_NoPrivateKey()
         {
-            ECParameters ecdsaKey = EccTestData.GetNistP256ReferenceKey();
-
-            // no private key
             byte[] compositeKey = ComposeKeys(
                 MLDsaTestsData.IetfMLDsa65.PrivateSeed,
-                WriteECPrivateKey(version: 1, d: null, ecdsaKey.Curve.Oid.Value, point: ecdsaKey.Q));
+                WriteECPrivateKey(version: 1, d: null, oid: null, point: null));
 
             AssertImportBadPrivateKey(CompositeMLDsaAlgorithm.MLDsa65WithECDsaP256, compositeKey);
         }
 
         [Fact]
-        public static void ImportBadPrivateKey_ECDsa_WrongCurve()
+        public static void ImportBadPrivateKey_ECDsa_HasCurve()
         {
-            CompositeMLDsaAlgorithm algorithm = CompositeMLDsaAlgorithm.MLDsa65WithECDsaP256;
-
-            // Wrong curve OID
-            AssertImportBadPrivateKey(
-                algorithm,
-                CreateKeyWithCurveOid(ECCurve.NamedCurves.nistP521.Oid.Value));
+            ECParameters ecdsaKey = EccTestData.GetNistP256ReferenceKey();
 
+            // Domain parameters are not allowed
             AssertImportBadPrivateKey(
-                algorithm,
-                CreateKeyWithCurveOid("1.3.36.3.3.2.8.1.1.7")); // brainpoolP256r1
-
-            // Domain parameters are optional, don't throw (unless platform does not support Composite ML-DSA)
-            CompositeMLDsaTestHelpers.AssertImportPrivateKey(
-                import => AssertThrowIfNotSupported(() => import(), algorithm),
-                algorithm,
-                CreateKeyWithCurveOid(ECCurve.NamedCurves.nistP256.Oid.Value));
-
-            static byte[] CreateKeyWithCurveOid(string? oid)
-            {
-                ECParameters ecdsaKey = EccTestData.GetNistP256ReferenceKey();
-
-                return ComposeKeys(
+                CompositeMLDsaAlgorithm.MLDsa65WithECDsaP256,
+                ComposeKeys(
                     MLDsaTestsData.IetfMLDsa65.PrivateSeed,
-                    WriteECPrivateKey(version: 1, ecdsaKey.D, oid, ecdsaKey.Q));
-            }
+                    WriteECPrivateKey(version: 1, ecdsaKey.D, ecdsaKey.Curve.Oid.Value, point: null)));
         }
 
         [Fact]
-        public static void ImportPrivateKey_ECDsa_NoPublicKey()
+        public static void ImportPrivateKey_ECDsa_HasPublicKey()
```

Comment:
Finger in the wind, it feels far more likely that a key would have both than "either".  So in addition to the two negative tests here, may as well make a third one which verifies that we're not suddenly stunned into acceptance by "both".